tests/tasks/test_binary_container_build.py

```diff
@@ -142,17 +147,18 @@ def mock_workflow_data(*, enabled_platforms: List[str]) -> inner.ImageBuildWorkf
     return mocked_data
 
 
-def mock_config(registry_config: Dict[str, Any], remote_hosts_config: Dict[str, Any],
+def mock_config(registry_config: List[Dict[str, Any]], remote_hosts_config: Dict[str, Any],
                 image_size_limit=0):
     """Make load_config() return mocked config.
 
     The registry property of the mocked config will return the specified registry_config.
     The remote_hosts property will return the remote hosts config.
     """
-    cfg = config.Configuration()
-    flexmock(cfg).should_receive("registry").and_return(registry_config)
-    flexmock(cfg).should_receive("remote_hosts").and_return(remote_hosts_config)
-    flexmock(cfg).should_receive("image_size_limit").and_return({'binary_image': image_size_limit})
+    raw_config = {'version': 1,
+                  ReactorConfigKeys.REGISTRIES_KEY: registry_config,
+                  ReactorConfigKeys.REMOTE_HOSTS_KEY: remote_hosts_config,
+                  ReactorConfigKeys.IMAGE_SIZE_LIMIT_KEY: {'binary_image': image_size_limit}}
+    cfg = config.Configuration(raw_config=raw_config)
     flexmock(BinaryBuildTask).should_receive("load_config").and_return(cfg)
```

Comment:
You could return the config here and then you wouldn't have to create a Task instance in test_get_authfile_path(), you could do something like this
```python
cfg = mock_config(...)
assert get_authfile_path(cfg.registry) == ...
```

Reply:
good idea! updated accordingly